scripts/NGI_pipeline_test.py

```diff
@@ -0,0 +1,238 @@
+
+
+
+
+def touch(file):
+    open(file, "w").close()
+    
+    
+def create_sample_name(project_identifier, sample_number, facility):
+    sample_id = str(sample_number).zfill(3)
+    if facility == 'stockholm':
+        return "{}_{}".format(project_identifier, sample_id)
+    else:
+        return "U{}".format(sample_id)
+    
+        
+
+
+
+def writeProjectData(data, token, base_url):
+    project=data
+    samples=project.pop('samples', None)
+    url=base_url+'/api/v1/project'
+    projson=json.dumps(project)
+    writeToCharon(projson, url, token)
+    for sid in samples:
+        libs=samples[sid].pop('libs', None)
+        sampjson=json.dumps(samples[sid])
+        url=base_url+'/api/v1/sample/'+project['projectid']
+        writeToCharon(sampjson, url, token)
+        for lib in libs:
+            seqruns=libs[lib].pop('seqruns', None)
+            libjson=json.dumps(libs[lib])
+            url=base_url+'/api/v1/libprep/'+project['projectid']+"/"+sid
+            writeToCharon(libjson, url, token)
+            for seqrun in seqruns:
+                seqjson=json.dumps(seqruns[seqrun])
+                url=base_url+'/api/v1/seqrun/'+project['projectid']+"/"+sid+"/"+lib
+                writeToCharon(seqjson, url, token)
+
+
+def writeToCharon(jsonData, url, token):
+    session = requests.Session()
+    headers = {'X-Charon-API-token': token, 'content-type': 'application/json'}
+    r=session.post(url, headers=headers, data=jsonData)
+
+
+def cleanCharon(pid, url, token):
+    session = requests.Session()
+    headers = {'X-Charon-API-token': token, 'content-type': 'application/json'}
+    r=session.delete(url+'/api/v1/project/'+pid, headers=headers)
+
+
+
+def create_proj_name(project_name=None, facility='stockholm'):
+    if project_name == None:
+        if facility == 'stockholm':
+            project_name_header = "J.Doe"
+            year = generate_random_numbers(2)
+            pnum = generate_random_numbers(2)
+            project_name = "{}_{}_{}".format(project_name_header, year, pnum)
+        else:
+            project_name = "{}-{}".format(generate_random_chars(2), generate_random_numbers(4))
+    else:
+        project_name = project_name
+
+    if facility == 'stockholm':
+        project_id = "P{}".format(generate_random_numbers(3))
+    else:
+        project_id = project_name
+    #TODO: CHECK THAT THE PROJECT DOES NOT EXISTS ALREADY
+    return (project_name, project_id)
+
+
+def create_fake_FC():
+    # create something like 160217_ST-E00201_0063_AHJHNYCCXX
+    new = False
+    while (not new):
+        # date
+        date = generate_random_numbers(6)
+        # instrument name
+        instrument = generate_random_instrument()
+        # run id
+        run_id = generate_random_numbers(4)
+        # FC position
+        FC_pos = random.choice(["A", "B"])
+        # FC name
+        FC_name = generate_random_chars(9)
```

Comment:
Just to be on the safe side for downstream checks in the pipeline, I would do something like `FC_name = generate_random_chars(7) + 'XX'`